x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -162,20 +165,34 @@ export const createLifecycleExecutor =
       ActionGroupIds
     > = {
       alertWithLifecycle: ({ id, fields }) => {
+        const alert = alertFactory.create(id);
```

Comment:
Switch the order of these calls so that the alert is not added to `currentAlerts` if `alertFactory.create` throws an error.